pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1912,12 +1912,14 @@ func buildListener(opts buildListenerOpts, trafficDirection core.TrafficDirectio
 				}
 			}
 			if !fullWildcardFound {
-				sort.Strings(chain.sniHosts)
+				chain.sniHosts = append([]string{}, chain.sniHosts...)
```

Comment:
what effect does this have? looks like its appending a slice to an empty slice and assigning it to the original slice

Reply:
Before: chain.sniHosts points to some point in memory storing the slice. Many other chain.sniHosts also point to it. This results in calling sort.Strings on the same array concurrently which is not legal
After: chain.sniHosts points to some point in memory storing the slice. We copy that slice into our own point in memory. Many other chain.sniHosts also have a slice, but each on is its own copy. sort.Strings is no longer called concurrently on the same slice.

Reply:
ah, makes sense. thanks!

Reply:
@howardjohn - I see the code is running in a loop, but don't see it in a go routine, so how is it concurrent?

Reply:
The assumption being made is that its being modified somewhere else concurrently.

See https://github.com/istio/istio/issues/28861#issuecomment-727056641.

Reply:
Cool, thanks! 
Thanks @howardjohn for the fix.